tests/tpch/generate-data.py

```diff
@@ -1,25 +1,80 @@
+import functools
 import pathlib
 import tempfile
+import warnings
 
 import botocore.session
 import click
+import coiled
 import dask
 import duckdb
 import psutil
+import pyarrow.compute as pc
+from dask.distributed import wait
 
 
 def generate(
-    scale: int = 10, partition_size: str = "50 MiB", path: str = "./tpch-data"
+    scale: int = 10,
+    partition_size: str = "50 MiB",
+    path: str = "./tpch-data",
+    relaxed_schema: bool = False,
 ):
     if str(path).startswith("s3"):
         path += "/" if not path.endswith("/") else ""
-        path += f"scale-{scale}/"
+        path += f"scale-{scale}{'-strict' if not relaxed_schema else ''}/"
+        use_coiled = True
     else:
-        path = pathlib.Path(path) / f"scale-{scale}/"
+        path = (
+            pathlib.Path(path)
+            / f"scale-{scale}{'-strict' if not relaxed_schema else ''}/"
+        )
         path.mkdir(parents=True, exist_ok=True)
+        use_coiled = False
 
     print(f"Scale: {scale}, Path: {path}, Partition Size: {partition_size}")
 
+    if use_coiled:
+        with coiled.Cluster(
+            n_workers=10,
+            worker_memory="4 GiB",
+            worker_options={"nthreads": 1},
+            region="us-east-2",
+        ) as cluster:
+            cluster.adapt(minimum=1, maximum=250)
+            client = cluster.get_client()
+
+            jobs = []
+            for step in range(0, scale):
+                job = client.submit(
+                    _dbgen_to_path,
+                    scale=scale,
+                    step=step,
+                    path=path,
+                    relaxed_schema=relaxed_schema,
+                    partition_size=partition_size,
+                )
+                jobs.append(job)
+            wait(jobs)
+    else:
+        return _dbgen_to_path(scale, path, partition_size, relaxed_schema)
+
+
+def retry(f):
+    @functools.wraps(f)
+    def _(*args, **kwargs):
+        for _ in range(5):
+            try:
+                return f(*args, **kwargs)
+            except Exception as exc:
+                warnings.warn(f"Failed w/ {exc}, retrying...")
+                continue
+        return f(*args, **kwargs)
+
+    return _
+
+
+@retry
+def _dbgen_to_path(scale, path, partition_size, relaxed_schema, step=None):
```

Comment:
This could probably use a better name.  Maybe also a brief docstring.  I had to puzzle over what this did for a while.  (it's late for me and my brain isn't working well, which is probably a good target reader persona to consider when writing)

Reply:
Not great, I've renamed and added to it a bit. If it could be better, please let me know!